src/ppx/graphics_util.cpp

```diff
@@ -31,6 +32,178 @@
+uint32_t GetPlaneHeightInPixels(
+    const grfx::FormatPlaneDesc::Plane& plane,
+    grfx::FormatChromaSubsampling       subsampling,
+    uint32_t                            imageHeight)
+{
+    bool hasColSubsampling = (subsampling == grfx::FORMAT_CHROMA_SUBSAMPLING_420);
+    bool hasChromaValue    = false;
+    bool hasLumaValue      = false;
+    for (auto it = plane.members.begin(); it != plane.members.end(); ++it) {
+        const grfx::FormatPlaneDesc::Member& member = *it;
+        if (member.type == grfx::FORMAT_PLANE_CHROMA_TYPE_CHROMA) {
+            hasChromaValue = true;
+        }
+        else if (member.type == grfx::FORMAT_PLANE_CHROMA_TYPE_LUMA) {
+            hasLumaValue = true;
+        }
+        else {
+            PPX_LOG_WARN("Member " << member.component << "has unknown chroma type.");
+        }
+    }
+
+    if (hasColSubsampling && hasChromaValue) {
+        // Note: you never have subsampling on the height axis of the image in
+        // a plane if luma values are present, since luma values usually aren't
+        // subsampled. You might have subsampling on the width axis, but that
+        // would essentially mean you get two luma values, and one of each
+        // chroma value, in a block of four.
+        if (hasLumaValue) {
+            PPX_LOG_WARN(
+                "Frame size will be inaccurate, there is vertical subsampling "
+                "with both chroma and luma values present on a single plane, "
+                "which is not supported!");
+        }
+
+        // If we're subsampling at 4:2:0, the image will have half its height.
+        return imageHeight / 2;
+    }
+    return imageHeight;
+}
+
+// Gets the width of a single plane, in terms of number of pixels represented.
+// This doesn't directly correlate to the number of bits / bytes for the plane's
+// height. The value returned can be used in a copy-image-to-buffer command.
+// plane: The plane to get the width for (containing information about the
+//        color components represented in the plane).
+// subsampling: The type of subsampling applied to chroma values for the image
+//              (e.g. 444, 422, 420).
+// imageWidth: The width of the image, in pixels, with no subsampling applied.
+uint32_t GetPlaneWidthInPixels(
+    const grfx::FormatPlaneDesc::Plane& plane,
+    grfx::FormatChromaSubsampling       subsampling,
+    uint32_t                            imageWidth)
+{
+    bool hasRowSubsampling = (subsampling == grfx::FORMAT_CHROMA_SUBSAMPLING_420) ||
+                             (subsampling == grfx::FORMAT_CHROMA_SUBSAMPLING_422);
+    bool hasChromaValue = false;
+    for (auto it = plane.members.begin(); it != plane.members.end(); ++it) {
+        const grfx::FormatPlaneDesc::Member& member = *it;
+        if (member.type == grfx::FORMAT_PLANE_CHROMA_TYPE_CHROMA) {
+            hasChromaValue = true;
+            break;
+        }
+    }
```

Comment:
not really a bit deal, but could simply the code a bit with `std::any_of`, something like:
```
bool hasChromaValue = std::any_of(plane.members.begin(), plane.members.end(), 
                       [](const grfx::Member& m) { 
                           return m.type == grfx::FORMAT_PLANE_CHROMA_TYPE_CHROMA; 
                       });
```

Reply:
Is functional-style preferred in bigwheels? I can update all relevant loops if so!

Reply:
good question, I am not sure if there is a specific style we should follow, @apazylbe: is there any preference on this in BW? thanks

Reply:
There are no clear preferences. However, be mindful of the readability:
Here, it's quite clear: any_of and a simple equal. IMO as readable as the equivalent range-based for if not more.
But if you start to convert all range-based for into `std::for_each`, then it can become messier than the simpler alternative.

My rule of thumb is: if the lambda is a one-liner, and I can use a `std::accumulate`, `std::any` or this kind of wrapper, I use it. If it's just to replace a for with a std::for_each, I don't.

Reply:
For sure - I was referring to the other loops of this form in this class

Reply:
Yeah, the syntax is a bit messy. If you feel like it's less readable, stick with classic loops 😊